### PR TITLE
Initial project, code and data for reproducing CNTK hang issue

### PR DESCRIPTION
**Pull request by mdabros**

### CODE REVIEW

**Comment by mdabros** [comment]
@nietras Can you have a quick view and see if this should be sufficient for reproducing the hang issue. I have not succeeded in reproducing the issue using the code yet, but it is currently still running.

File: src/CntkSerializerIssue/Program.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            {
+                { "Channel1", Path.Combine(repositoryRoot, @"mapfiles\TrainChannel1.map") },
+                { "Channel2", Path.Combine(repositoryRoot, @"mapfiles\TrainChannel2.map") },
+                { "Channel3", Path.Combine(repositoryRoot, @"mapfiles\TrainChannel3.map") },
+                { "Channel4", Path.Combine(repositoryRoot, @"mapfiles\TrainChannel4.map") },
+            };
+
+            var ctfFilePath = Path.Combine(repositoryRoot, @"mapfiles\TrainTargets.ctf");
+            var outputShape = 3;
+            var maxSweeps = int.MaxValue;
```
Comment (nietras) [inline src/CntkSerializerIssue/Program.cs:30]: @mdabros I think we use a special value for infinite sweeps in the place we have the issue, probably not important but it is a difference.

File: src/CntkSerializerIssue/Program.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                {
+                    System.Console.WriteLine("Current sweep: " + sweeps);
+                }
+
+                var minibatchData = source.GetNextMinibatch(minibatchSize, d);
+
+                // Stop training once max epochs is reached.
+                if (minibatchData.empty())
+                {
+                    System.Console.WriteLine("Completed sweeps");
```
Comment (nietras) [inline src/CntkSerializerIssue/Program.cs]: nit: completed a sweep not "sweeps"

Comment (nietras) [reply src/CntkSerializerIssue/Program.cs]: nit: write number

File: mapfiles/TrainTargets.ctf [hunk 3]
```diff
@@ -0,0 +1,128 @@
+|targets 0 0 10
```
Comment (nietras) [inline mapfiles/TrainTargets.ctf:1]: @mdabros no mask, not sure needed, but that is a difference too

**Review by nietras — COMMENTED** [review]
Overall this looks very similar to what we do when we have the hang issue. See other comments for what might be different.

File: src/CntkSerializerIssue/Program.cs (hunk 1 above)
Comment (nietras) [reply src/CntkSerializerIssue/Program.cs:30]: @mdabros note this comment about maxSweeps...

**Comment by nietras** [comment]
@mdabros would have been good if you had committed the images first in separate PR to separate out the 1000s of files ;)

File: src/CntkSerializerIssue/Program.cs (hunk 1 above)
Comment (mdabros) [reply src/CntkSerializerIssue/Program.cs:30]: @nietras I checked, we seem to use `ulong.MaxValue;` for testing, and `totalNumberOfEpochs` for training. So this should be quite similar.

File: mapfiles/TrainTargets.ctf (hunk 3 above)
Comment (mdabros) [reply mapfiles/TrainTargets.ctf:1]: @nietras If it is not possible to reproduce the issue using without the mask, I will try to add it, and see if it makes a difference.

**Comment by mdabros** [comment]
Yes, it would have been much preferable with a separate PR for the images files :-)

**Pull request merged**